### ISSUE TITLE
Using WebView leads to size error in platform_views since Flutter 3.3.0

### ISSUE THREAD

**Issue by gerken-tss** [flutter/flutter#111255]
Since the Flutter upgrade to 3.3.0 I am unable to use a WebView (neither [flutter_webview](https://pub.dev/packages/webview_flutter) nor third party packages like [flutter_inappwebview](https://pub.dev/packages/flutter_inappwebview)) as it leads to an assertion error from `TextureAndroidViewController`.

I have a strong suspicion that it's caused by https://github.com/flutter/flutter/pull/101399.

This issue happens if parts of a `Scaffold` like the `BottomNavigationBar` have a varying height which is only known after a certain amount of time (asynchronously). This can be the case e. g. if menu items are loaded via an API.

## Steps to Reproduce

1. Paste the below code sample in a fresh Flutter 3.3.0 project
2. Make sure to have a sufficient `minSdkVersion` (I used `23`)
3. Execute `flutter run` on the code sample with Flutter version of at least 3.0.0 
4. Wait 100 ms
5. Realize that the WebView is not rendered and the terminal issues an error output

**Expected results:** 
The WebView is rendered properly and the `BottomNavigationBar` is visible.

**Actual results:** 
There is an error and neither the WebView nor the `BottomNavigationBar` is displayed.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Scaffold(
        body: WebView(
          initialUrl: 'https://flutter.dev',
        ),
        bottomNavigationBar: BottomNavigationExample(),
      ),
    );
  }
}

class BottomNavigationExample extends StatefulWidget {
  const BottomNavigationExample({
    Key? key,
  }) : super(key: key);

  @override
  State<BottomNavigationExample> createState() =>
      _BottomNavigationExampleState();
}

class _BottomNavigationExampleState extends State<BottomNavigationExample> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return FutureBuilder(
      future: Future.delayed(const Duration(milliseconds: 100)),
      builder: (context, snapshot) {
        if (!snapshot.hasData) {
          return Container();
        }

        return BottomNavigationBar(
          items: const <BottomNavigationBarItem>[
            BottomNavigationBarItem(icon: Icon(Icons.add), label: 'Test1'),
            BottomNavigationBarItem(icon: Icon(Icons.add), label: 'Test2'),
          ],
        );
      },
    );
  }
}

```

</details>

<details>
  <summary>Logs</summary>

```
I/flutter (19778): ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
I/flutter (19778): │ #0   _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
I/flutter (19778): │ #1   _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
I/flutter (19778): │ #2   TextureAndroidViewController._sendCreateMessage (package:flutter/src/services/platform_views.dart:1129:12)
I/flutter (19778): │ #3   AndroidViewController.create (package:flutter/src/services/platform_views.dart:792:11)
I/flutter (19778): │ #4   _PlatformViewLinkState.build.<anonymous closure> (package:flutter/src/widgets/platform_view.dart:876:22)
I/flutter (19778): │ #5   _PlatformViewPlaceholderBox.performLayout (package:flutter/src/widgets/platform_view.dart:1115:13)
I/flutter (19778): │ #6   RenderObject.layout (package:flutter/src/rendering/object.dart:2135:7)
I/flutter (19778): │ #7   RenderBox.layout (package:flutter/src/rendering/box.dart:2418:11)
I/flutter (19778): ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
I/flutter (19778): │ ⛔ [null] 'package:flutter/src/services/platform_views.dart': Failed assertion: line 1129 pos 12: '!size.isEmpty': trying to create TextureAndroidViewController without setting a valid size.
```

```
Analyzing test-app...                               
No issues found! (ran in 8.1s)

```

```
[✓] Flutter (Channel stable, 3.3.0, on macOS 12.5.1 21G83 darwin-x64, locale de-DE)
    • Flutter version 3.3.0 on channel stable at /Users/marcgerken/dev/flutter-sdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffccd96b62 (10 days ago), 2022-08-29 17:28:57 -0700
    • Engine revision 5e9e0e0aa8
    • Dart version 2.18.0
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/marcgerken/Library/Android/sdk/
    • Platform android-33, build-tools 30.0.3
    • ANDROID_HOME = /Users/marcgerken/Library/Android/sdk/
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F100
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.70.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

**Comment by darshankawar** [comment]
@gerken-tss 
Thanks for the detailed report. Can you try the same on latest master and see if you get same behavior ?

`flutter channel master`
`flutter upgrade`

**Comment by gerken-tss** [comment]
@darshankawar 
Thank your for the quick response!

I was able to reproduce the above mentioned behavior on channel `master`:

```
[✓] Flutter (Channel master, 3.4.0-19.0.pre.161, on macOS 12.5.1 21G83 darwin-x64, locale de-DE)
    • Flutter version 3.4.0-19.0.pre.161 on channel master at /Users/marcgerken/fvm/versions/master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ad946f8478 (18 minutes ago), 2022-09-09 05:24:36 -0700
    • Engine revision a71e3a5480
    • Dart version 2.19.0 (build 2.19.0-177.0.dev)
    • DevTools version 2.17.0
```

**Comment by darshankawar** [comment]
Thanks for the update. Using the code sample and verifying on latest master and stable, I am getting same error log.

<details>
<summary> stable, master flutter doctor -v </summary>

```

[✓] Flutter (Channel stable, 3.3.1, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.3.1 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4f9d92fbbd (3 days ago), 2022-09-06 17:54:53 -0700
    • Engine revision 3efdf03e73
    • Dart version 2.18.0
    • DevTools version 2.15.0

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[✓] Flutter (Channel master, 3.4.0-19.0.pre.156, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.4.0-19.0.pre.156 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 857116da40 (26 minutes ago), 2022-09-09 00:37:25 -0400
    • Engine revision 6610f3f2a9
    • Dart version 2.19.0 (build 2.19.0-177.0.dev)
    • DevTools version 2.17.0
    
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by gerken-tss** [comment]
Is it somehow related to https://github.com/flutter/flutter/issues/110146?

**Comment by SalsabeelaHasan** [comment]
any solution?

` :::::::: runZonedGuarded: Caught error in my root zone. ERROR : 'package:flutter/src/services/platform_views.dart': Failed assertion: line 1129 pos 12: '!size.isEmpty': trying to create TextureAndroidViewController without setting a valid size. :::::::
`

**Comment by gerken-tss** [comment]
@SalsabeelaHasan 
Either you downgrade your Flutter version or you provide a static height for your BottomNavigationBar or whatever widget is causing your trouble. For me, it always occurred when during loading time (fetching data via API) I returned a different sized widget than after the request has completed. Does this work for your application?

**Comment by SalsabeelaHasan** [comment]
Thanks for your suggestion @gerken-tss .
Which version of flutter does not have this problem? to do a downgrade.
For the second suggestion, I do not use the BottomNavigationBar widget on the same page with webviwe, how can I determine which widget is the cause?

**Comment by gerken-tss** [comment]
@gerken-tss I guess everything before 3.3.0 should be fine. For testing purpose I can recommend [fvm](https://fvm.app/). This helped me narrow down the issue. Regarding finding out which widget is the root cause, I know no better way than commenting out widgets one by one until the issue does not persist anymore. In my case it was only caused by Scaffold widgets such as AppBar and BottomNavigationBar. Would be interesting to hear which widget is the cause in your case!

**Comment by SalsabeelaHasan** [comment]
Many thanks! @gerken-tss .
I tried to finger out the cause, by commenting out widgets one by one, in my case there is no problem in the scaffold widget, the problem does not persist anymore when I set a static height for the container that holds the webviwe widget instead of a dynamic one (_updatedHeight variable).
 This is an annoying problem! I need to update the height of webviwe for many reasons... 

<img width="840" alt="Screen Shot 2022-09-27 at 2 57 30 AM" src="https://user-images.githubusercontent.com/62775384/192401037-b3193f92-ede5-49c4-999a-a312fe964fc6.png">

**Comment by SalsabeelaHasan** [comment]
Hello @stuartmorgan,
 I see your actions here, is there any solution that will go to be out soon? 
 thanks.

**Comment by stuartmorgan** [comment]
This will be fixed on `master` when the PR linked above lands.

**Comment by SalsabeelaHasan** [comment]
@stuartmorgan ، Is there an expected date for this update to be released?

**Comment by stuartmorgan** [comment]
It depends on what you mean by "released". It will be available in the `master` channel as soon as it lands; https://github.com/flutter/flutter/wiki/Flutter-build-release-channels explains the release cadence for other channels.

**Comment by SalsabeelaHasan** [comment]
Thanks for your help @stuartmorgan .
What I mean is, when will I be able to use a version with the solution, I have tasks and features that have been disrupted by the presence of this bug.
If you have any suggestion that helps me at the moment, please let me know.

**Comment by SalsabeelaHasan** [comment]
Hi @darshankawar ,
is the fix introduced on the 3.3.4 version that released in Oct 05 ?
